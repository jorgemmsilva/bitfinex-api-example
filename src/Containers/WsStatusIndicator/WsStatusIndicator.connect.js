import { connect } from 'react-redux'
import { getWebsocketIsConnecting, getWebsocketIsConnected } from '../../Store/websocket/selectors'
import WsStatusIndicator from './WsStatusIndicator'

export const mapState = state => ({
  connecting: getWebsocketIsConnecting(state),
  connected: getWebsocketIsConnected(state),
})

export default connect(mapState)(WsStatusIndicator)
