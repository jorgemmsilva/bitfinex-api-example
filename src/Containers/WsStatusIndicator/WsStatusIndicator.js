import React from 'react'
import PropTypes from 'prop-types'
import {
  IndicatorWrapper, ConnectingIcon, ConnectedIcon, DisconnetedIcon,
} from './styled'

const WsStatusIndicator = ({ connecting, connected }) => {
  let icon = <ConnectingIcon />
  if (!connecting) {
    icon = connected ? <ConnectedIcon /> : <DisconnetedIcon />
  }
  return (
    <IndicatorWrapper>
      { icon }
    </IndicatorWrapper>
  )
}

WsStatusIndicator.propTypes = {
  connecting: PropTypes.bool.isRequired,
  connected: PropTypes.bool.isRequired,
}

export default WsStatusIndicator
