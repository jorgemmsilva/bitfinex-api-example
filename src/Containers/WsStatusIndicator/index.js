export { default } from './WsStatusIndicator.connect'
