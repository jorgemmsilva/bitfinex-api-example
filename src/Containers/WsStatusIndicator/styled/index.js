import styled, { keyframes } from 'styled-components'
import { Autorenew, Check, Close } from '@material-ui/icons'

export const IndicatorWrapper = styled.div`
  position: absolute;
  right: 10px;
  top: 0;
  border: 1px solid black;
  border-top: 0;
  border-radius: 0px 0px 7px 7px;
  padding-top: 2px;
`

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`
export const ConnectingIcon = styled(Autorenew)`
  animation: ${rotate} 2s linear infinite
`

export const ConnectedIcon = styled(Check)`
  color: green
`

export const DisconnetedIcon = styled(Close)`
  color: red
`
