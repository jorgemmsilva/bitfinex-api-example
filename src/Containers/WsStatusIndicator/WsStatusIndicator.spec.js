import React from 'react'
import { shallow } from 'enzyme'
import WsStatusIndicator from './WsStatusIndicator'
import {
  ConnectingIcon, ConnectedIcon, DisconnetedIcon,
} from './styled'

it('renders correctly when connecting', () => {
  const wrapper = shallow(<WsStatusIndicator connecting connected={false} />)
  expect(wrapper.find(ConnectingIcon).length).toBe(1)
  expect(wrapper).toMatchSnapshot()
})


it('renders correctly when connected', () => {
  const wrapper = shallow(<WsStatusIndicator connecting={false} connected />)
  expect(wrapper.find(ConnectedIcon).length).toBe(1)
  expect(wrapper).toMatchSnapshot()
})

it('renders correctly when disconnected', () => {
  const wrapper = shallow(<WsStatusIndicator connecting={false} connected={false} />)
  expect(wrapper.find(DisconnetedIcon).length).toBe(1)
  expect(wrapper).toMatchSnapshot()
})
