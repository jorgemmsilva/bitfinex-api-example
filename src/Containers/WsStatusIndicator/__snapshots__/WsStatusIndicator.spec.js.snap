// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when connected 1`] = `
<styled.div>
  <Styled(pure(CheckIcon)) />
</styled.div>
`;

exports[`renders correctly when connecting 1`] = `
<styled.div>
  <Styled(pure(AutorenewIcon)) />
</styled.div>
`;

exports[`renders correctly when disconnected 1`] = `
<styled.div>
  <Styled(pure(CloseIcon)) />
</styled.div>
`;
